optional-clustering-analysis/clustering-report-template.Rmd

```
@@ -0,0 +1,174 @@
+library(scater)
+```
+
+```{r}
+# Read in SCE object
+processed_sce <- read_rds(file.path(params$processed_sce))
+```
+
+## Plots
+
+## Clustering results for `r params$library`
+
+Below is a UMAP plot where each cell has been colored based on the cell's cluster assignment. 
+Here clustering was performed using `r params$cluster_type` clustering with a nearest neighbors value of `r params$nearest_neighbors`.
+
+```{r}
+# If param$nearest_neighbors_range is a single value, set nearest_neighbors_increment to NULL
+if(length(params$nearest_neighbors_range) == 1) {
+  nearest_neighbors_increment <- 1
+}
+# If a nearest neighbors increment exists, then create a sequence for clustering 
+if(!(params$nearest_neighbors_increment == 1)){
+  nn_range <- seq(min(params$nearest_neighbors_range), 
+                  max(params$nearest_neighbors_range), 
+                  params$nearest_neighbors_increment) 
+  # If no nearest neighbors increment has been input then the provided range is 
+  # directly used for clustering 
+} else {
+  nn_range <- params$nearest_neighbors_range
+}
+
+# Grab column name with clustering results
+cluster_column_names <- paste(params$cluster_type, nn_range, sep = "_")
+existing_columns <- intersect(cluster_column_names, colnames(colData(processed_sce)))
+
+# Plot
+plotReducedDim(processed_sce, dimred = "UMAP", colour_by = existing_columns) +
+  theme_bw() +
+  labs(
+    caption = paste0(
+      stringr::str_to_title(params$cluster_type),
+      " clustering with nearest neighbors value ",
+      params$nearest_neighbors_range
+    )
+  ) +
+  guides(col = guide_legend("Cluster assignment")) +
+  theme(text = element_text(size = 14),
+        plot.caption = element_text(hjust = 0.5))
+```
+
+## Plot cluster validity stats
+
+```{r fig.height = 9.5}
+# Grab the cluster validity stats data frame
+all_validity_stats_df <- readr::read_tsv(file.path(
+  params$stats_dir,
+  paste0(params$library, "_clustering_all_validity_stats.tsv")
+))
```

After reading this in, I would add a check that makes sure that data corresponding to the `existing_clusters` is present and if it's not there I would throw an error. I also would probably subset the dataframe to those `existing_clusters` cluster names. This should apply to all of the dataframes.